optuna/trial/_trial.py

```diff
@@ -673,18 +674,28 @@ def _is_relative_param(self, name: str, distribution: BaseDistribution) -> bool:
         return distribution._contains(param_value_in_internal_repr)
 
     def _check_distribution(self, name: str, distribution: BaseDistribution) -> None:
-        old_distribution = self._cached_frozen_trial.distributions.get(name, distribution)
+        old_distribution = self._cached_frozen_trial.distributions.get(name, None)
+
+        if old_distribution is None:
+            return
+
         if old_distribution != distribution:
```

Comment:
As the check `old_distribution != distribution` can be expensive during a study, I would suggest this change.

```suggestion
        distribution_changed = old_distribution != distribution
        if distribution_changed:
```